.ci/Jenkinsfile

```diff
@@ -445,26 +446,90 @@ stage('Build') {
     }
     finally {
         stage('Merge Artifacts') {
-            node(pCloud) {
-                checkout scm  // checking out the SCM so the coverage report can load the source
-                builds.each { item ->
-                    copyArtifacts(
-                        projectName: item.fullProjectName,
-                        selector: specific("${item.number}"),
-                        fingerprintArtifacts: true,
-                        optional: true,
-                    )
+            String yaml = """
+apiVersion: v1
+kind: Pod
+spec:
+    volumes:
+      - emptyDir:
+            medium: Memory
+        name: dshm
+    containers:
+      - name: pod
+        image: ${lintImage}
+        imagePullPolicy: Always
+        command: ["sleep"]
+        args: [${pLintTimeout}]
+        volumeMounts:
+          - mountPath: /dev/shm
+            name: dshm
+        resources:
+            requests:
+                cpu: ${lintCpuLimit}
+                ephemeral-storage: ${lintMemLimit}
+                memory: ${lintMemLimit}
+            limits:
+                cpu: ${lintCpuLimit}
+                ephemeral-storage: ${lintMemLimit}
+                memory: ${lintMemLimit}
+"""
+            def podArgs = [
+                cloud: pCloud,
+                inheritFrom: pCloud,
+                label: env.BUILD_TAG,
+                yaml: yaml,
+                nodeSelector: nodeSelector,
+            ]
+            echo "pod args: $podArgs"
+            podTemplate(podArgs) {
+                node(env.BUILD_TAG) {
+                    container('pod') {
+                        checkout scm  // checking out the SCM so the coverage report can load the source
+
+
+                        builds.each { item ->
+                            copyArtifacts(
+                                projectName: item.fullProjectName,
+                                selector: specific("${item.number}"),
+                                fingerprintArtifacts: true,
+                                optional: true,
+                            )
+                        }
+
+                        sh "mkdir -p $buildOutputFolder"
+
+                        String mergedCoverageFile = "$buildOutputFolder/merged.coverage.xml"
+
+                        sh """#!/usr/bin/env bash
```

Comment:
[nit] Similar comment to above, not a huge fan of embedding bash scripts as strings in groovy.  Would be nice if we just converted this to a util script or something and called it vs. embedding this in functional code.

Reply:
Good point; done.